analyzers/src/SonarAnalyzer.CSharp/Trackers/CSharpObjectInitializationTracker.cs

```diff
@@ -172,8 +172,8 @@ private bool ObjectCreatedWithAllowedValue(ObjectCreationExpressionSyntax object
             }
             else
             {
-                // if no tracked properties are being explicitly set or passed as arguments, the constructor may still be safe by design
-                return constructorIsSafe;
+                // if no tracked properties are being explicitly set or passed as arguments, check if the default constructor is safe
+                return isDefaultConstructorSafe != null ? isDefaultConstructorSafe() : constructorIsSafe;
```

Comment:
I want to challenge `constructorIsSafe` field and constructor argument. It looks redundant & confusing now. Is it really still needed in this class now? With the new parameters, I'd say it should not be here.